server/src/utils/multipart.rs

```diff
@@ -174,6 +180,42 @@ fn consume_stream(
     Box::new(future)
 }
 
+//        //check if the data is actually an embedded multipart field (Electron linux hack)
+//        let boundary = get_multipart_boundary(data.as_ref());
+//
+//        let boundary = get_multipart_boundary(data.as_ref());
+//
+//        if let Some(boundary) = boundary {
+//            let multipart = Multipart::new(
+//                Ok(boundary.to_string()),
+//                futures::stream::once(Ok(Bytes::from(data))),
+//            );
+//            return consume_stream(content, multipart);
+//        }
+
+/// Looks for a multipart boundary at the beginning of the data
+/// and returns it as a `&str` if it is found
+///
+/// A multipart boundary starts at the beginning of the data (possibly
+/// after some blank lines) and it is prefixed by '--' (two dashes)
+///
+/// ```ignore
+/// let boundary = get_multipart_boundary(b"--The boundary\r\n next line");
+/// assert_eq!(Some("The boundary"), boundary);
+///
+/// let invalid_boundary = get_multipart_boundary(b"The boundary\r\n next line");
+/// assert_eq!(None, invalid_boundary);
+/// ```
+pub fn get_multipart_boundary(data: &[u8]) -> Option<&str> {
+    data.split(|&byte| byte == b'\r' || byte == b'\n')
+        // Get the first non-empty line
+        .find(|slice| !slice.is_empty())
+        // Check for the form boundary indicator
+        .filter(|slice| slice.len() > 2 && slice.starts_with(b"--"))
```

Comment:
the length check should not be necessary, as `"".starts_with("--")` just returns `false`.

Reply:
I need to check that the line "--\n" is invalid if I just check starts_with(b"--") it would be considered valid.